src/test/testserver.rs

```diff
@@ -57,12 +57,12 @@ impl TestServer {
                     eprintln!("testserver: handling just-accepted stream: {}", e);
                     break;
                 }
-                thread::spawn(move || handler(stream.unwrap()));
-                if done.load(Ordering::Relaxed) {
+                if done.load(Ordering::SeqCst) {
```

Comment:
I didn't see a a mention of this change in the PR description. What was it for? I'm curious about the choice of `SeqCst` vs `Relaxed`; I didn't give it much thought when I originally wrote this code.

Reply:
I got intermittent test errors saying connection closed. I couldn't repro consistently and decided for a belts are bracers.

Reply:
I'm working on merging the recent timeout fixes into `release-2.0` and am also getting connection closed errors. I believe these are actually caused because `dribble_headers_respond` does not properly finish its headers with `\r\n`, so ureq sees that as a connection that was closed while headers were still being read. The underlying issue is that `dribble_headers_respond` shouldn't be getting to the end of writing its headers at all; we should time out before then. I've fixed the test so it emits a clearer error in that case, and am working on figuring out why it's reaching the end of headers (i.e. why it's not timing out).